### PR TITLE
Add burst mode for weapons and fix Stormdog ammo with it

### PR DESCRIPTION
**Pull request by JonnyH**
Add burst mode weapons and fix the stormdog ammunition as part of that

### CODE REVIEW

**Comment by JonnyH** [comment]
Untested rebase of #758

File: game/state/gamestate.cpp [hunk 1]
```diff
@@ -654,31 +654,37 @@ void GameState::fillPlayerStartingProperty()
...
 	}*/
+
 	for (auto &pair : this->initial_vehicles)
 	{
 		auto v = current_city->createVehicle(*this, pair.first, this->getPlayer(), {this, bld});
 		v->homeBuilding = v->currentBuilding;
 		for (auto &eq : pair.second)
 		{
-			v->addEquipment(*this, eq);
+			auto device = v->addEquipment(*this, eq);
```
Comment (JonnyH) [inline game/state/gamestate.cpp]: This is setting all the initial vehicle ammo to full? I've noticed that before, good to have it fixed - but it should probably be a different change?

File: game/state/gamestate.cpp [hunk 2]
```diff
@@ -654,31 +654,37 @@ void GameState::fillPlayerStartingProperty()
...
+			device->ammo = eq->max_ammo;
 		}
 	}
+
 	// Give the player initial vehicle equipment
 	for (auto &pair : this->initial_vehicle_equipment)
 	{
 		base->inventoryVehicleEquipment[pair.first.id] = pair.second;
 	}
+
```
Comment (JonnyH) [inline game/state/gamestate.cpp]: These extra lines aren't really needed, and if you really think they make it clearer it should be a separate change anyway

File: game/state/gamestate.cpp (hunk 1 above)
Comment (FilmBoy84) [reply game/state/gamestate.cpp]: Yes, it's been a bug in OA for a while
Wasn't sure if to separate or not as it relates to the weapon this PR is changing for burst mode
If you want it separate i can do that?
I have to change the burst from 10 to 3 anyway

File: game/state/gamestate.cpp (hunk 2 above)
Comment (FilmBoy84) [reply game/state/gamestate.cpp]: Sorry, which two lines? The two blank ones in the section above or all the changes to gamestate.cpp

If just the blank ones, then agreed, they are not needed im happy for them to be removed

Comment (JonnyH) [reply game/state/gamestate.cpp]: The blank ones - separating out logical changes into different commits (or even PRs) often makes things easier going forward - when doing stuff like merges or rebases like this, for example :)

So each "single change", be it bug fix, feature, or formatting change, ideally should each be in different changes.

**Comment by FilmBoy84** [comment]
Referencing this in PR #790 which carries an updated version of this PR
Once 790 is merged, this PR can also be closed without merge

**Pull request closed without merging**